src/librustc/macros.rs

```diff
@@ -83,7 +83,12 @@ macro_rules! __impl_stable_hash_field {
 macro_rules! impl_stable_hash_for {
     // FIXME(mark-i-m): Some of these should be `?` rather than `*`. See the git blame and change
     // them back when `?` is supported again.
-    (enum $enum_name:path { $( $variant:ident $( ( $($field:ident $(-> $delegate:tt)*),* ) )* ),* $(,)* }) => {
+    (enum $enum_name:path {
+        $( $variant:ident
+           $( ( $($field:ident $(-> $delegate:tt)*),* ) )*
+           $( { $($named_field:ident $(-> $named_delegate:tt)*),* } )*
```

Comment:
This will accept something like `Variant(a, b, c){ d, e, f }`, right? It doesn't seem like a real problem, but just to clarify. (If it does, a comment pointing this out would be nice.)

Reply:
Yup - I'll add a comment.